packages/vitest/src/utils/traces.ts

```diff
@@ -235,12 +235,34 @@ export class Traces {
     return tracer.startSpan(name, options, context)
   }
 
+  // On browser mode, async context is not automatically propagated,
+  // so we manually bind the `$` calls to the provided context.
+  // TODO: this doesn't bind to user land's `@optelemetry/api` calls
```

Comment:
Should we provide a way to inherit it? (Expose this function _somehow_?) Maybe not for this PR

Reply:
I think so. I've thought of some ideas. One approach is that we could implement and provide own context manager (and also use internally as default), so that they'll ensure root tester context. For example, if we can assume such context manager, then we don't need this `bind` trick. Also we might be able to enhance custom context manager to be aware of per-test function context (assuming they are non concurrent tests).

I'll create a follow up issue for this.